packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -83,8 +83,8 @@ export class CoffeeCompiler extends CachingCompiler {
 
     let sourceMap = JSON.parse(output.v3SourceMap);
 
-    if (source.match(/(`|import|export|function\*)/)) {
-      // If source contains backticks or ES2015+ features,
+    if (source.match(/`|\bimport\b|\bexport\b|\byield\b/)) {
```

Comment:
> import or export by definition need to be at the beginning of a line, which in CoffeeScript means they follow a newline (no semicolons)

Not common, but CoffeeScript actually _can_ have semi-colons.  But based on your description, my second regex above [should work](https://regex101.com/r/xP9lB8/1) but would need the `/m` modifier.  Again though, this only eliminates some _potential_ false positives.

> I can’t imagine it that takes more time to run this match than it does to recompile with Babel. Assuming most projects don’t use features that generate ES2015 code, which I think is a pretty safe assumption, this regex should be saving us time. And most projects that do generate ES2015 probably are doing so because of modules, which means this regex finds a match almost immediately, from the top of the file, so the match should be a very minor performance hit.

Agree, with the same assumptions.  Though I'd still recommend the first regex change regardless.